ui-v2/app/templates/dc/services/show/upstreams.hbs

```diff
@@ -0,0 +1,32 @@
+<div id="upstreams" class="tab-section">
+  <div role="tabpanel">
+  {{#if (gt gateway.Services.length 0)}}
+    <section>
+      <p>
+        Upstreams are services that may recieve traffic from this gateway. Learn more about configuring gateways in our
+        <a href={{env 'CONSUL_INGRESS_GATEWAYS_URL'}} target="_blank">documentation.</a>
+      </p>
+      <ListCollection @cellHeight={{73}} @items={{gateway.Services}} class="consul-gateway-services-list" as |item index|>
+        <a data-test-service-name href={{href-to 'dc.services.show' item.Name}}>
+          {{item.Name}}
+        </a>
+        <ul>
+        {{#if (not-eq item.GatewayConfig.ListenerPort 0)}}
+          <li class="port">
+            <CopyButton
+              @value={{item.GatewayConfig.ListenerPort}}
+              @name="Address"
+            />
+            <span>:{{item.GatewayConfig.ListenerPort}}</span>
+          </li>
+        {{/if}}
+        </ul>
+      </ListCollection>
+    </section>
+  {{else}}
+    <p>
+      There are no linked services.
```

Comment:
Is this just copy/pasta'ed from the link Services tab, should this be changed to say upstreams?

Reply:
Ops, yes. Good catch 😉.